src/js/common/parser.js

```diff
@@ -0,0 +1,223 @@
+/*
```

Comment:
We chatted briefly, really we are just talking about exposing the path to the error in the schema in addition to the description, so that it is possible to wire in an alternate feedback system.  That system would be expected to make sense of the schema and the path provided on its own.

I also have been thinking about adding a grade that helps associate the feedback with onscreen elements using a system of bindings and/or rules as done in previous work.